### ISSUE TITLE
[Project 3] Question about run_list in real time scheduler

### ISSUE THREAD

**Issue by mushypeas** [swsnu/osspr2021#84]
In the implemented scheduler entity for the real time scheduler(`sched_rt_entity`), there is a list_head field named `run_list`. I've been looking through the implementation of the scheduler in rt.c for some time, but couldn't get a clear concept of the usage of the `run_list`. Why is `run_list` necessary in each `task_struct`, when the `rt_prio_array` in the run queue(`rt_rq`) is keeping track of all the tasks?

Thank you!

**Comment by mushypeas** [comment]
아 run_list 가 여러 node를 저장하는 list가 아니라 해당 프로세스를 가리키는, 단순히 rt_prio_array에 들어갈 list_node인 건가요?

**Comment by WoosukKwon** [comment]
Yes, it is. `rt_prio_array` maintains 100 queues, each of which is a list of `sched_rt_entity`s with the same priority. Technically, each queue in `rt_prio_array` is a `list_head`, and `sched_rt_entity`s are added to (or deleted from) the list through their `run_list` fields. For example, in  `__enqueue_rt_entity`, you can see that a `sched_rt_entity` is enqueued by `list_add(&rt_se->run_list, queue)`. Likewise, in `pick_next_rt_entity`, you can check that the next `sched_rt_entity` to run is selected by `next = list_entry(queue->next, struct sched_rt_entity, run_list)`.

**Comment by mushypeas** [comment]
답변 감사합니다!
